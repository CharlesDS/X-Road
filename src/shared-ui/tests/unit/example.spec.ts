// Libraries
import Vuetify from 'vuetify';

// Components
import LargeButton from '@/components/LargeButton.vue';

// Utilities
import { createLocalVue, mount } from '@vue/test-utils';

describe('LargeButton.vue', () => {
  const localVue = createLocalVue();
  let vuetify: Vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('renders props.msg when passed', () => {
    const msg = 'Test';
    const wrapper = mount(LargeButton, {
      localVue,
      vuetify,
      propsData: { msg },
    });

    // Create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();
  });
});
