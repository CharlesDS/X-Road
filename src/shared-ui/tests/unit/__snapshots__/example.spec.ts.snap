// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeButton.vue renders props.msg when passed 1`] = `<button type="button" class="large-button v-btn v-btn--contained v-btn--rounded theme--light v-size--default primary" style="min-width: 120px;" msg="Test"><span class="v-btn__content"></span></button>`;
